src/Components/Forms/src/EditContextDataAnnotationsExtensions.cs

```diff
@@ -67,7 +67,10 @@ private void OnFieldChanged(object? sender, FieldChangedEventArgs eventArgs)
 
                     Validator.TryValidateProperty(propertyValue, validationContext, results);
                     _messages.Clear(fieldIdentifier);
-                    _messages.Add(fieldIdentifier, results.Select(result => result.ErrorMessage!));
+                    foreach (var result in CollectionsMarshal.AsSpan(results))
```

Comment:
Is this faster than `foreach (var result in results)` because it skips something like checking-whether-the-list-was-modified on each read?

If so, then as a meta-point about this feature itself rather than your use of it here, it seems like a weird way of expressing the intent because there's nothing obvious in the naming of `CollectionsMarshal.AsSpan` that implies this, or provides any indication of what would go wrong if the assumption turned out to be false. Perhaps this is something we only expect advanced .NET devs to do and don't feel it needs to be made any more obvious.

Reply:
It's this "unsafe" way of getting you access to the internal array. The *Marshal is a pattern .NET uses to express that these APIs are advanced and could do bad things if misused

Reply:
Yup - it basically gives you access to the static array that the list is using as it's implementation: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/CollectionsMarshal.cs#L19. It's only safe to use if the list isn't modified as part of the iteration which is true in this particular case.